ESP_socket_embedded_motors/src/main.cpp

```diff
@@ -0,0 +1,616 @@
+        if (heartBeatNum > 4)
+        {
+            heartBeatNum = 1;
+        }
+    }
+
+    if (millis() - lastMotorFeedback >= 2000) // Change to 1000?
+    {
+        lastMotorFeedback = millis();
+        motorFeedback();
+    }
+
+    // Safety timeout
+    safety_timeout();
+
+    // Motor status debug printout
+#if defined(DEBUG2)
+    if (millis() - lastMotorStatus > 500) {
+        lastMotorStatus = millis();
+
+        // Status 1
+        Serial.print(millis() - Motor2.status1.timestamp);
+        Serial.print(" ms ago: ");
+        Serial.print(Motor2.status1.motorTemperature);
+        Serial.print(" *C; ");
+        Serial.print(Motor2.status1.busVoltage);
+        Serial.print(" V; ");
+        Serial.print(Motor2.status1.outputCurrent);
+        Serial.print(" A; ");
+        Serial.print(Motor2.status1.sensorVelocity);
+        Serial.println(" RPM");
+        
+        // Status 2
+        Serial.print(millis() - Motor2.status2.timestamp);
+        Serial.print(" ms ago: ");
+        Serial.print(Motor2.status2.sensorPosition);
+        Serial.println(" rotations");
+    }
+#endif
+
+    //-------------//
+    //  CAN Input  //
+    //-------------//
+
+    static CanFrame rxFrame;
+    if (ESP32Can.readFrame(rxFrame, 1)) {
+        // Decode the ID
+
+        uint32_t msgId = rxFrame.identifier;
+        // Pull out device ID and API ID
+        uint8_t deviceId = msgId & 0x3F;
+        uint32_t apiId = (msgId >> 6) & 0x3FF;
+
+        if (apiId == 0x60) {
+            for (int i = 0; i < MOTOR_AMOUNT; i++) {
+                if (deviceId == motorList[i]->getID()) {
+                    motorList[i]->parseStatus0(rxFrame.data);
+                    break;
+                }
+            }
+        } else if (apiId == 0x61) {  // Status 1
+            for (int i = 0; i < MOTOR_AMOUNT; i++) {
+                if (deviceId == motorList[i]->getID()) {
+                    motorList[i]->parseStatus1(rxFrame.data);
+                    break;
+                }
+            }
+        } else if (apiId == 0x62) {  // Status 2
+            for (int i = 0; i < MOTOR_AMOUNT; i++) {
+                if (deviceId == motorList[i]->getID()) {
+                    motorList[i]->parseStatus2(rxFrame.data);
+                    break;
+                }
+            }
+        } else if ((apiId & 0x300) == 0x300) {  // Parameter
+#ifdef DEBUG_STATUS
+            printREVFrame(rxFrame);
+#endif
+            Serial.print("Got parameter ");
+            Serial.print(apiId & 0xFF, HEX);
+            Serial.print(" for: ");
+            for (int i = 0; i < MOTOR_AMOUNT; i++) {
+                if (deviceId == motorList[i]->getID()) {
+                    Serial.print(i);
+                }
+            }
+            Serial.print(" (type ");
+            Serial.print(rxFrame.data[4]);
+            Serial.print("): ");
+            //  uint32_t
+            if (rxFrame.data[4] == static_cast<uint8_t>(sparkMax_ParameterType::kUint32)) {
```

Comment:
again, what does this mean / do? should have comments saying what each of these bytes are for